### ISSUE TITLE
ImageCleaner bug with data from CSV

### ISSUE THREAD

**Issue by stmandl** [fastai/fastai#2344]
When following [lesson2-download](https://github.com/fastai/course-v3/blob/master/nbs/dl1/lesson2-download.ipynb) in the fast.ai course-v3, I ran into a problem, when using the ImageCleaner to remove duplicate images. I get this error message:

```text
AttributeError: 'numpy.str_' object has no attribute 'is_file'
```

I'm using fastai that's built from the github master branch.
This code reproduces the problem:

```python
from fastai.vision import *
from fastai.widgets import *

path = untar_data(URLs.MNIST_TINY)
data_from_csv = ImageList.from_csv(path, csv_name='labels.csv').split_none().label_from_df().transform(get_transforms(), size=224).databunch()
learn_cln = cnn_learner(data_from_csv, models.resnet34, metrics=error_rate)
ds, idxs = DatasetFormatter().from_similars(learn_cln)
ImageCleaner(ds, idxs, path=path, duplicates=True)
```

I also have a trivial fix for the problem, but as I'm a total fast.ai newbie (I started using it today), I wanted to make sure that it is an actual bug in the library code and not an error on my part using the library.

**Installation details**

```text
=== Software === 
python        : 3.7.3
fastai        : 1.0.58.dev0
fastprogress  : 0.1.21
torch         : 1.2.0
nvidia driver : 430.26
torch cuda    : 10.0.130 / is available
torch cudnn   : 7602 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 7979MB | GeForce RTX 2070

=== Environment === 
platform      : Linux-5.0.0-29-generic-x86_64-with-debian-buster-sid
distro        : Ubuntu 18.04 bionic
conda env     : base
python        : /home/stm/anaconda3/bin/python
sys.path      : 
/home/stm/anaconda3/lib/python37.zip
/home/stm/anaconda3/lib/python3.7
/home/stm/anaconda3/lib/python3.7/lib-dynload
/home/stm/anaconda3/lib/python3.7/site-packages
/home/stm/fastai/fastai

Sun Sep 22 22:52:06 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 430.26       Driver Version: 430.26       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 2070    Off  | 00000000:08:00.0  On |                  N/A |
| 29%   27C    P8     9W / 175W |   3197MiB /  7979MiB |      2%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1387      G   /usr/lib/xorg/Xorg                           172MiB |
|    0      1522      G   /usr/bin/gnome-shell                         146MiB |
|    0      2285      G   ...3791373277734,131072 --gpu-preferences=   329MiB |
|    0      2653      C   /home/stm/anaconda3/bin/python              2543MiB |
+-----------------------------------------------------------------------------+

```

**Comment by sgugger** [comment]
I think your fix is the correct one, you should submit a PR with it. Thanks for flagging the issue!

**Comment by sgugger** [comment]
Fixed by #2347

**Comment by pseudorandomcoder** [comment]
I had this error when importing labels from cleaned.csv. Changed code in "image_cleaner.py" for a quick fix:
        idxs = ((i for i in fns_idxs if items[i].is_file())
changed to:
        idxs = ((i for i in fns_idxs if Path(items[i]).is_file())
